### PR TITLE
add fake-lodash-pure-js types

### PR DESCRIPTION
**Pull request by gxless**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by twksos — APPROVED** [review]

File: types/fake-lodash-pure-js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for fake-lodash-pure-js 1.0
+// Project: https://github.com/gxless/fake-lodash-pure-js
+// Definitions by: Xiang Gao <https://github.com/gxless>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+export function head<T>(array: T[]): T | undefined;
+export function tail<T>(array: T[]): T | undefined;
+export function isArray(input: any): boolean;
```
Comment (peterblazejewicz) [inline types/fake-lodash-pure-js/index.d.ts:9]: is this OK to return boolean value, when function behind implementation seems to return only `T | undefined` for array like or just `undefined` for everything else?
https://github.com/gxless/fake-lodash-pure-js/blob/1c1aa16ddd0ebf6877049b55975d89888737f15d/src/services/head.js
https://github.com/gxless/fake-lodash-pure-js/blob/main/src/index.js#L2-L4

Comment (gxless) [reply types/fake-lodash-pure-js/index.d.ts:8]: fixed in https://github.com/gxless/fake-lodash-pure-js/blob/main/src/index.js#L2-L4
thanks for your review!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@gxless thx!

**Comment by gxless** [comment]
Ready to merge

**Pull request merged**